examples/platform/nxp/se05x/DeviceAttestationSe05xCredsExample_v2.cpp

```diff
@@ -172,49 +173,50 @@ CHIP_ERROR ExampleSe05xDACProviderv2::SignWithDeviceAttestationKey(const ByteSpa
 
     ChipLogDetail(Crypto, "Sign using DA key from se05x (Using internal sign)");
 
-    MutableByteSpan tlv_buffer = ((MutableByteSpan &) message_to_sign).SubSpan(1); // Exclude the Start container
-
     TLV::TLVReader msg_reader;
-    TLV::Tag tagValue;
+    TLV::TLVReader tagReader;
 
-    msg_reader.Init(tlv_buffer);
-    msg_reader.Next();
+    msg_reader.Init(message_to_sign);
 
     /* To be removed. Use common key id to sign message */
     static bool sign_cert_decl_attest = 1;
 
     if (sign_cert_decl_attest)
     {
-        /* Skip certificate declaration tag */
-        msg_reader.Next();
-
-        tagValue = msg_reader.GetTag();
-        VerifyOrReturnError(TLV::ContextTag(2) == tagValue, CHIP_ERROR_INVALID_TLV_TAG);
-        /* Get attestation nonce */
-        ByteSpan attest_nonce;
-        msg_reader.Get(attest_nonce);
-        /* Set attestation nonce */
-        VerifyOrReturnError(CHIP_NO_ERROR ==
-                                se05xSetCertificate(CD_ATTEST_NONCE_DATA_SE05X_ID, attest_nonce.data(), attest_nonce.size()),
-                            CHIP_ERROR_INTERNAL);
-
-        msg_reader.Next();
-        tagValue = msg_reader.GetTag();
-        VerifyOrReturnError(TLV::ContextTag(3) == tagValue, CHIP_ERROR_INVALID_TLV_TAG);
-        ByteSpan time_stamp;
-        msg_reader.Get(time_stamp);
-        uint8_t tmp = time_stamp.size();
+        /* Get length and Skip certificate declaration tag */
+        ReturnErrorOnFailure(TLV::Utilities::Find(msg_reader, TLV::ContextTag(1), tagReader));
+        uint8_t cdlen = tagReader.GetLength();
+
+        ReturnErrorOnFailure(TLV::Utilities::Find(msg_reader, TLV::ContextTag(2), tagReader));
+        uint8_t attlen = tagReader.GetLength();
+        if (attlen > 0)
+        {
+            /* Get attestation nonce */
+            ByteSpan attest_nonce;
+            ReturnErrorOnFailure(tagReader.Get(attest_nonce));
+            /* Set attestation nonce */
+            VerifyOrReturnError(CHIP_NO_ERROR ==
+                                    se05xSetCertificate(CD_ATTEST_NONCE_DATA_SE05X_ID, attest_nonce.data(), attest_nonce.size()),
+                                CHIP_ERROR_INTERNAL);
+        }
+
+        ReturnErrorOnFailure(TLV::Utilities::Find(msg_reader, TLV::ContextTag(3), tagReader));
+        uint8_t tslen = tagReader.GetLength();
         /* Set time stamp length */
-        VerifyOrReturnError(CHIP_NO_ERROR == se05xSetCertificate(CD_TIME_STAMP_LEN_SE05X_ID, &tmp, 1), CHIP_ERROR_INTERNAL);
-        if (time_stamp.size() > 0)
+        VerifyOrReturnError(CHIP_NO_ERROR == se05xSetCertificate(CD_TIME_STAMP_LEN_SE05X_ID, &tslen, 1), CHIP_ERROR_INTERNAL);
```

Comment:
Note that if the TLV is malicious then `tslen` might have an arbitrary value here...  your should really not use it for anything important until the `Get()` for the `ByteSpan` succeeds.

Reply:
Get() would fail in case the length in TLV is 0. But i would still need the length information to be passed to HSM. So not relying on the status of Get() here. ALso the data created in HSM is transient data. So this should be fine.

Reply:
No, my point is that the `tslen` might be large enough it's larger than the TLV buffer, so this call returns a garbage value and then `Get` fails

Reply:
Changed the sequence of calls now. Using length after get() is called.